src/main/java/com/vmware/vipclient/i18n/VIPCfgFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.i18n;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class VIPCfgFactory {
+
+    /**
+     * A lazy-loaded singleton VIPCfg using the "initialization-on-demand holder" design pattern
+     */
+    private static class GlobalCfgHolder {
+        static final VIPCfg globalCfg = new VIPCfg();
+    }
+
+    public static VIPCfg getGlobalCfg() {
+        return GlobalCfgHolder.globalCfg;
+    }
+
+    /**
+     * A lazy-loaded singleton Map<String, VIPCfg> configs using the "initialization-on-demand holder" design pattern
+     */
+    private static class ConfigsHolder {
+        static final Map<String, VIPCfg> configs = new ConcurrentHashMap<>();
+    }
+
+    public static VIPCfg getCfg(String productName) {
+        if (!contains(productName)) {
+            synchronized (ConfigsHolder.configs) {
+                if (!contains(productName)) {
+                    VIPCfg cfg = new VIPCfg();
+                    cfg.setProductName(productName);
+                }
+            }
+        }
+        return ConfigsHolder.configs.get(productName);
+    }
+
+    static void addCfg(VIPCfg cfg) {
```

Comment:
Make this private and call this in line 35. In this way, only one method takes charge of writing config map. This is clear.

Reply:
By call cfg.setProductName(productName) with different product names, users can save the same config instance multiple time with different keys. Is this a problem?

Reply:
If you check cfg.setProductName, it has been deprecated and it only sets the deprecated VIPCfg.productName for backwards compatibility. It does not add the instance to configs map.